node/chainSimulator/chainSimulator.go

```diff
@@ -0,0 +1,111 @@
+package chainSimulator
+
+import (
+	"errors"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/endProcess"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/configs"
+)
+
+const (
+	genesisAddressWithStake = "erd10z6sdhwfy8jtuf87j5gnq7lt7fd2wfmhkg8zfzf79lrapzq265yqlnmtm7"
+	//genesisAddressWithStakeSK   = "ZWRlZDAyNDczZTE4NjQ2MTY5NzNhZTIwY2IzYjg3NWFhM2ZmZWU1NWE2MGQ5NDgy\nMjhmMzk4ZTQ4OTk1NjA3NTc4YjUwNmRkYzkyMWU0YmUyNGZlOTUxMTMwN2JlYmYy\nNWFhNzI3NzdiMjBlMjQ4OTNlMmZjN2QwODgwYWQ1MDg="
+	genesisAddressWithBalance = "erd1rhrm20mmf2pugzxc3twlu3fa264hxeefnglsy4ads4dpccs9s3jsg6qdrz"
+	//genesisAddressWithBalanceSK = "YWQxMTM2YTEyNWZkM2YxY2ZiMTU0MTU5NDQyZTRiYzZhM2I1YzMwOTU5NDMwMjk5\nNThhYzQ2NGRhN2NlMTNlYjFkYzdiNTNmN2I0YTgzYzQwOGQ4OGFkZGZlNDUzZDU2\nYWI3MzY3Mjk5YTNmMDI1N2FkODU1YTFjNjIwNTg0NjU="
+)
+
+type simulator struct {
+	chanStopNodeProcess    chan endProcess.ArgEndProcess
+	syncedBroadcastNetwork components.SyncedBroadcastNetworkHandler
+	nodes                  []ChainHandler
+	numOfShards            uint32
+}
+
+func NewChainSimulator(numOfShards uint32, pathToInitialConfig string) (*simulator, error) {
+	if pathToInitialConfig == "" {
```

Comment:
I would omit this check. Reason? Someone might place the binary in the same config directory.

Reply:
removed